src/pages/postgraphile/jwk-verification.md

```diff
@@ -1,184 +1,205 @@
+Instead, we want to follow the pattern of Postgraphile and return errors in
+the format our consumers expect: a JSON response. 
 
-```javascript{31-66}
-const jwt = require("jsonwebtoken");
-// ...
-app.use(
-  postgraphile(process.env.DATABASE_URL, process.env.DB_SCHEMA, {
-    pgSettings: async req => {
-      try {
-        const authHeader = req.headers.authorization;
-        if (!authHeader) {
-          throw new Error("No authorization header provided.");
-        }
-
-        const authSplit = authHeader.split(" ");
-
-        if (authSplit.length !== 2) {
-          throw new Error(
-            'Malformed authentication header. "Bearer accessToken" syntax expected.'
-          );
-        } else if (authSplit[0].toLowerCase() !== "bearer") {
-          throw new Error(
-            '"Bearer" keyword missing from front of authorization header.'
-          );
-        }
-
-        const token = authSplit[1];
-        const decodedToken = jwt.decode(token, { complete: true });
-
-        if (decodedToken === null) {
-          throw new Error("Unable to decode JWT, refresh login and try again.");
-        }
-
-        // Follow Auth0 tutorial linked above for this function flow
-        // (particularly around `expressJwtSecret`).
-        const secretKey = await getSecretKey(
-          req,
-          decodedToken.header,
-          decodedToken.payload
-        );
-
-        // Once the matching secret key is retrieved from the Auth0 provided
-        // JWK, we can verify the token against it.
-        const jwtClaims = await jwt.verify(
-          token,
-          secretKey.publicKey,
-          {
-            audience: process.env.AUTH0_AUDIENCE,
-            issuer: `https://${process.env.AUTH0_TENANT}.auth0.com/`,
-            algorithms: ["RS256"],
-          },
-          function(err, verifiedToken) {
-            if (err) {
-              throw new Error(err);
-            }
-          }
-        );
-
-        // You can perform any final verification of the jwtClaims here and
-        // throw an error if they're invalid.
-
-        // Finally return the transaction settings to use.
-        // See: https://graphile.org/postgraphile/usage-library/#pgsettings-function
-        return {
-          /* e.g. */
-          // 'role': jwtClaims.role,
-          // 'jwt.claims.user_id': jwtClaims.user_id,
-          // ...
-        };
-      } catch (e) {
-        e.status = 401; // append a generic 401 Unauthorized header status
-        throw e;
-      }
-    },
-  })
-);
+Let's create a basic Express middleware for handling the errors which
+our `checkJwt` function will throw:
+
+```
+const authErrors = (err, req, res, next) => {
+  if (err.name === 'UnauthorizedError') {
+    console.log(err); // You will still want to log the error...
+    // but we don't want to send back internal operation details
+    // like a stack trace to the client!
+    res.status(err.status).json({ error: err.message }); 
```

Comment:
This should probably return the error in a GraphQL compatible format; i.e.`{errors:[{message: err.message}]}`

Reply:
To match the spec! Of course, apologies for overlooking that.